Optimizations

McJezuss opened the pull request:
# Pre-release commit
This is the pre-release commit into the staging branch 😃 

There are a lot of commits made here (more than was planned), but I think it is finally ready to be merged into the staging branch.

## Things that were added that drastically improves the QOL is:
- Showing that you're on the wrong network
- If the user changes networks, the app does not crash

## The different pages revisited:
- ### Home
  * When the user visits the page for the first time, the project info will be shown at the top, but from the next visit, the leaderboard will be shown at the top.
  * The leaderboard shows the correct rankings with the participant's nickname if they set it. If they did not, it will show their address instead.
  
- ### The Journey
    * It now tracks the milestones as set by the Jackpot contract. I omitted Genesis purposely.
    * Adding milestones to the Jackpot contract will automatically update the page.
    * I am considering adding the NFT of the matching milestone to the block as an "incentive" on desktop only. (Tell me your thoughts on this)
    * The current milestone icon is shown in a different colour compared to the other ones.

- ### Refuel
    * Refuel shows the user's balance if they have some volume to refuel with. When we launch on an exchange I will add a "Buy Volume" button that will redirect them to the exchange.
 
- ### Profile
   * There is now a user profile page that is accessed by clicking on your address after you connected your wallet.
   * Here users can claim their nickname and change it if they want to.
   * It shows a Blockies seeded with their address as their profile picture. (We could make it so you can bind an NFT as your picture for a small VOL cost that will be burned every time you change it)
   * It shows a full breakdown of each milestone and the fuel you added to it.
   * It also tracks directly from the Jackpot contract, so as we add milestones it will start showing it here.
   * Owned and claimable NFTs will be shown here once they become available.

McJezuss committed:
Fixed fuel initial value

McJezuss committed:
Renamed details to refuel

McJezuss committed:
Renamed details to refuel

McJezuss committed:
Implemented page switches - disabled

McJezuss committed:
Optimized imports

McJezuss committed:
Moved background out to top layer + added refuel layout

McJezuss committed:
Added refuel function skeleton

humanshield89 committed:
Don't reload canvas on page changes

McJezuss committed:
Added refuel function

McJezuss committed:
Added input check to refuel

McJezuss committed:
Fixed dot crash

McJezuss committed:
Mobile optimization implemented

McJezuss committed:
Removed newspace

McJezuss committed:
Added material lab

McJezuss committed:
Fixed loading

McJezuss committed:
Removed old stats

McJezuss committed:
Fixed loading

McJezuss committed:
Added padding to top

McJezuss committed:
Added chainId

McJezuss committed:
Added rpcUrl

McJezuss committed:
Added revisit change

McJezuss committed:
Fixed sorting error

McJezuss committed:
Fixed double render issue

McJezuss committed:
Removed xs from container

McJezuss committed:
Moved WalletProvider to index

McJezuss committed:
Fixed spacing on topBar

McJezuss committed:
Removed react logo

McJezuss committed:
Added blockies

McJezuss committed:
Added profile + username claiming skeleton

McJezuss committed:
Added the journey (without text)

McJezuss committed:
Updated contract address

McJezuss committed:
Fixed leaderboard to show user if > 10th place

McJezuss committed:
Fixed a wrong wallet read

McJezuss committed:
Updated block time

McJezuss committed:
Updated abi

humanshield89 committed:
Freeze material-ui lab to alpha-58

humanshield89 committed:
Merge branch 'optimizations' of github.com:4-Fun-Coins/volume-ui into optimizations

McJezuss committed:
Added rounding to big numbers

McJezuss committed:
Added nicknames to leaderboards

humanshield89 committed:
use hashrouter instead of browser router to be ipfs friendly

humanshield89 committed:
use hashrouter instead of browser router to be ipfs friendly

McJezuss committed:
Changed rounding to 4 decimals

McJezuss committed:
Removed log

humanshield89 committed:
use tabs for navbar menu

humanshield89 committed:
use tabs for navbar menu

McJezuss committed:
Fixed seconds/block

humanshield89 committed:
Merge branch 'optimizations' into improvements/topbar

McJezuss committed:
Added dark colour

McJezuss committed:
Removed ProfileDialog

McJezuss committed:
Finishing profile 1/2

McJezuss committed:
Added volume-jackpot-abi

McJezuss committed:
Updated addresses

McJezuss committed:
Finishing profile 2/2

McJezuss committed:
Added stats & milestones to Journey

McJezuss committed:
Handle wrong network

McJezuss committed:
Fetch currently active milestone

McJezuss committed:
Changed wording

McJezuss committed:
Merge branch 'staging' into optimizations

McJezuss committed:
Removed log

McJezuss committed:
Merge remote-tracking branch 'origin/optimizations' into optimizations

McJezuss committed:
Added user icon in front of address

McJezuss committed:
Removed wallet.account & log

humanshield89 committed:
move layout to the root level and add errors on wrong network

humanshield89 committed:
Merge optimizations

humanshield89 committed:
optimize background canvas objects speed and positions

humanshield89 committed:
Merge pull request #21 from 4-Fun-Coins/improvements/topbar

Improvements/topbar

humanshield89 reviewed (commented):
Merged some quality of life changes , moved the background with the canvas to the root of the hierarchy so it won't be reloaded on page changes (this change overrides the multiple Layout mechanism )
Reduced the speed of objects in the canvas by a third to give a chance to low spec devices 
changed the drawer on mobile to use the tabs instead of list of links

humanshield89 committed:
add missing images and fix manifest description

humanshield89 committed:
make topbar logo clickable

humanshield89 committed:
Fix usable of variables inside useEfffect hook causing leak

humanshield89 committed:
add a small bottom border to TopBar

humanshield89 committed:
fix mobile friendliness on direct refuel

humanshield89 committed:
add small space between buttons in top bar

humanshield89 committed:
add small space between buttons in top bar

humanshield89 committed:
create new and improved leaderboard

humanshield89 committed:
Merge branch 'optimizations' of github.com:4-Fun-Coins/volume-ui into optimizations

humanshield89 committed:
New Dashboard

1 - New dashboard and leaderboard
2 - Separate Landing page and the leaderboard page
3 - use PNG images for the canvas
4 - add animations to page loading

humanshield89 committed:
Import Roboto font (material-ui default font)

humanshield89 committed:
Create Home page

humanshield89 committed:
remove cookies consent

humanshield89 committed:
Fix scroll essues

humanshield89 committed:
make lg the max width for the statistics page

humanshield89 committed:
change Hero image file size , explicit with on img element and fix alignement

humanshield89 committed:
Fix Settings Context there is no light theme

humanshield89 committed:
Create new and improved Journey page

humanshield89 committed:
fix future blocktime estimation

humanshield89 committed:
Cleanup unused styles

humanshield89 committed:
Fix mobile width on journey page

humanshield89 committed:
Make the dashboard leaderboard use the general look and feel of the site

humanshield89 committed:
fix arrow animation

humanshield89 committed:
Cleanup and delete legacy code

humanshield89 committed:
Finish VolumeContext.js
Cleanup code and fix minor bugs

humanshield89 committed:
add NFT Market place page

McJezuss committed:
Updated text

McJezuss committed:
Updated text

McJezuss committed:
Updated text

McJezuss committed:
Updated text

McJezuss committed:
Updated title
